### ISSUE TITLE
Set up DAP

### ISSUE THREAD

**Issue by JeanMarie-TTS** [GSA-TTS/FAC#1014]

**Comment by JeanMarie-TTS** [comment]
Refer #741 regarding DAP implementation on the transition site FAC.gov

**Comment by asteel-gsa** [comment]
[reference](https://github.com/GSA-TTS/FAC-transition-site/pull/3/files)

It appears the only reference we need to add is
```
  <!-- We participate in the US government's analytics program. See the data at analytics.usa.gov. -->
  <script async type="text/javascript" src="https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js?agency=GSA&subagency=TTS,OROS" id="_fed_an_ua_tag"></script>
```

**Comment by JeanMarie-TTS** [comment]
@heymatthenry , I am thinking of introducing the following code in tamplates/base.html right after the netatags block:
```
{% if 'prod' in request.get_host|lower %}
  <!-- We participate in the US government's analytics program. See the data at analytics.usa.gov. -->
  <script async type="text/javascript" src="https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js?agency=GSA&subagency=TTS,OROS" id="_fed_an_ua_tag"></script>
{% endif %}

```
Is this a good way to introduce this piece of code only in a production environment? Is `peod` a good string to look for?

**Comment by asteel-gsa** [comment]
> @heymatthenry , I am thinking of introducing the following code in tamplates/base.html right after the metatags block:
> 
> ```
> {% if 'prod' in request.get_host|lower %}
>   
>   <script async type="text/javascript" src="https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js?agency=GSA&subagency=TTS,OROS" id="_fed_an_ua_tag"></script>
> {% endif %}
> ```
> 
> 1. Is this a good way to introduce this piece of code only in a production environment? Is `prod` a good string to look for?
> 2. I am assuming that implementing this in base.html will make it applicable to the entire FAC site. Valid assumption?
> 3. Can't think of a good way to test this.

JeanMarie,

I have pinged @jperson1 about this, and james has some good insight into how this can be done. Would you mind setting up a meeting for us three to discuss? Any time before 1p PST seems to be good for the both of us.

**Comment by asteel-gsa** [comment]
Per @jperson1 and @JeanMarie-TTS meeting, we have discussed the following:
**DAP will not be implemented on any page that has login credentials, or any session that is from a logged in session**
Reference: [Implement DAP](https://digital.gov/guides/dap/add-your-site-dap/)
```
The DAP script should only be applied to public-facing pages. Public-facing web pages are defined as those that can be accessed without any authentication or login, and are not part of an otherwise “privileged session.”

The DAP script tag should not be placed on pages visited during logged-in sessions. Notably, other seemingly “public” pages that can be accessed without authentication may also be part of privileged sessions; for example, a password reset page that is accessed by clicking a link in an email is not appropriate for DAP code because it assumes the visitor has the privilege of control over the email account used to provide the link. 
```

**Proposed solution:**
Within `backend/templates/` we would like to add a file in `includes/` that will be the script to enable DAP. Within that script, we will create a conditional that references `DEBUG ENV`. 

The following logic would be present:
```
if (debug.env = true) {
 # do not apply script
} else {
 # apply script
}
 ```

The purpose of this logic is to not do DAP logging on DEV, and only on STAGING and PRODUCTION environments. With the addition of the `include file` this script will only be executed on the home page of the application, i.e `fac-<env>.app.cloud.gov` and will not be executed after logging in, or passing any credentials.

**Comment by JeanMarie-TTS** [comment]
@jperson1 , can you please do a quick review of the branch where I've made the DAP changes. I need some help with testing as my docker env seems broken with login not working.

**Comment by jperson1** [comment]
@JeanMarie-TTS 

Hello! I was able to confirm that:

1. The component housing the DAP script (`DAP.html`) loads on all screens.
2. The DAP script runs only when a user is not logged in.
3. The DAP script runs only when Django config setting `DEBUG` is set to False (as it is in the staging and production environments). 

I did find something interesting when testing - the DAP script can be blocked by plugins such as AdBlock. Something to keep in mind when testing later on.

It is recommended to move the DAP include statement up a few lines into the end of the header in `base.html`, with the meta properties. But, it does work in both places. So, if keeping it where it is makes sense, then it's no problem.
